### ISSUE TITLE
Layout/TrailingWhitespace errors on trailing TABs

### ISSUE THREAD

**Issue by nilclass** [rubocop/rubocop#10553]
The `Layout/TrailingWhitespace` cop throws an exception when encountering lines with trailing **tab** character(s).

--------

## Expected behavior

The cop should report trailing whitespace, and remove it if `--auto-correct` is given.

## Actual behavior

The cop raises an exception (see below).

## Steps to reproduce the problem

1. Create a file with this content:
```
<<~X
	
X
```
2. Run rubocop without any special options or configuration (I'm only using the `--debug` flag)

The error I'm getting is this:
```An error occurred while Layout/TrailingWhitespace cop was inspecting /home/niklas/src/reproduce-rubocop-issue/issue.rb.
undefined method `length' for nil:NilClass
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:88:in `whitespace_is_indentation?'
```

<details>
<summary>Full trace</summary>

```
An error occurred while Layout/TrailingWhitespace cop was inspecting /home/niklas/src/reproduce-rubocop-issue/issue.rb.
undefined method `length' for nil:NilClass
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:88:in `whitespace_is_indentation?'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:79:in `process_line_in_heredoc'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:69:in `block in process_line'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/base.rb:342:in `correct'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/base.rb:127:in `add_offense'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:67:in `process_line'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:54:in `block in on_new_investigation'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:51:in `each'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:51:in `each_with_index'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/layout/trailing_whitespace.rb:51:in `on_new_investigation'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:153:in `block (2 levels) in invoke'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:153:in `block in invoke'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:153:in `each'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:153:in `invoke'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:84:in `investigate'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cop/team.rb:83:in `investigate'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:309:in `inspect_file'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:280:in `loop'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:249:in `do_inspection_loop'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:130:in `block in file_offenses'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:155:in `file_offense_cache'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:129:in `file_offenses'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:67:in `block in warm_cache'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:587:in `call_with_index'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:557:in `process_incoming_jobs'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:537:in `block in worker'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:528:in `fork'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:528:in `worker'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:519:in `block in create_workers'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:518:in `each'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:518:in `each_with_index'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:518:in `create_workers'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:457:in `work_in_processes'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:294:in `map'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/parallel-1.22.1/lib/parallel.rb:238:in `each'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:67:in `warm_cache'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/runner.rb:46:in `run'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli/command.rb:11:in `run'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli/environment.rb:18:in `run'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli.rb:71:in `run_command'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli.rb:78:in `execute_runners'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/lib/rubocop/cli.rb:47:in `run'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/exe/rubocop:12:in `block in <top (required)>'
/home/niklas/.rvm/rubies/ruby-2.7.5/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
/home/niklas/.rvm/gems/ruby-2.7.5/gems/rubocop-1.27.0/exe/rubocop:12:in `<top (required)>'
/home/niklas/.rvm/gems/ruby-2.7.5/bin/rubocop:23:in `load'
/home/niklas/.rvm/gems/ruby-2.7.5/bin/rubocop:23:in `<main>'
/home/niklas/.rvm/gems/ruby-2.7.5/bin/ruby_executable_hooks:22:in `eval'
/home/niklas/.rvm/gems/ruby-2.7.5/bin/ruby_executable_hooks:22:in `<main>'
```
</details>

## RuboCop version

```
$ rubocop -V
1.27.0 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 2.7.5 x86_64-linux)
```

**Comment by dvandersluis** [comment]
I've replicated this, I'll fix it.

**Comment by nilclass** [comment]
Awesome, thanks for addressing this so quickly!
